openmc/plotter.py

```diff
@@ -540,3 +602,279 @@ def _calculate_xs_elem_mat(this, types, temperature=294., cross_sections=None,
+        i = 0
+        for g in range(library.energy_groups.num_groups):
+            data[line, i: i + 2] = mgxs[line, g]
+            i += 2
+
+    return np.flipud(energy_grid), data
+
+
+def _calculate_mgxs_nuc_macro(this, types, library, orders=None,
+                              temperature=294.):
+    """Determines the multi-group cross sections of a nuclide or macroscopic
+    object
+
+    If the data for the nuclide or macroscopic object in the library is
+    represented as angle-dependent data then this method will return the
+    average cross section over all angles.
+
+    Parameters
+    ----------
+    this : {openmc.Nuclide, openmc.Macroscopic}
+        Object to source data from
+    types : Iterable of str
+        The type of cross sections to calculate; values can either be those
+        in openmc.PLOT_TYPES_MGXS
+    library : openmc.MGXSLibrary
+        MGXS Library containing the data of interest
+    orders : Iterable of Integral, optional
+        The scattering order or delayed group index to use for the
+        corresponding entry in types. Defaults to the 0th order for scattering
+        and the total delayed neutron data.
+    temperature : float, optional
+        Temperature in Kelvin to plot. If not specified, a default
+        temperature of 294K will be plotted. Note that the nearest
+        temperature in the library for each nuclide will be used as opposed
+        to using any interpolation.
+
+    Returns
+    -------
+    data : numpy.ndarray
+        Cross sections calculated at the energy grid described by energy_grid
+
+    """
+
+    # Check the parameters and grab order/delayed groups
+    if orders:
+        cv.check_iterable_type('orders', orders, Integral,
+                               min_depth=len(types), max_depth=len(types))
+    else:
+        orders = [None] * len(types)
+    for i, line in enumerate(types):
+        cv.check_type("line", line, str)
+        cv.check_value("line", line, PLOT_TYPES_MGXS)
+        if orders[i]:
+            cv.check_greater_than("order value", orders[i], 0, equality=True)
+
+    xsdata = library.get_by_name(this.name)
+
+    if xsdata is not None:
+        # Obtain the nearest temperature
+        t = (np.abs(xsdata.temperatures - temperature)).argmin()
+
+        # Get the data
+        data = np.zeros((len(types), library.energy_groups.num_groups))
+        for i, line in enumerate(types):
+            if 'fission' in line and not xsdata.fissionable:
+                continue
+            elif line == 'unity':
+                data[i, :] = 1.
+            else:
+                temp_data = getattr(xsdata, _PLOT_MGXS_ATTR[line])[t]
+                shape = temp_data.shape[:]
+                # If we have angular data, then we will plot the average
+                # over all provided angles.  Since the angles are equi-distant,
+                # un-weighted averaging will suffice
+                if xsdata.representation == 'angle':
```

Comment:
Perhaps it is just me, but I find this complex `if-else` logic confusing. I think it could use a few brief comments for any developers who may wish to maintain/extend this in the future.